### DISCUSSION TITLE
Support for using `find` method with json pointers

### DISCUSSION THREAD

**Discussion by Smakson** [nlohmann/json#4264] [Ideas]
Currently, one can use json pointer to check for whether a path is present in a json object via `contains` and one can access the path via `at` and `operator[]`. However, this effectively forces us to do the sequence of lookups along the path twice which forces the user that wants to use JSON pointer into an unnecessary performance pessimization.

From looking at the existing implementations of e.g. `contains` in the `json_pointer` class it seems like that would not be hard, in fact I am working on a branch/PR to propose an implementation, but I wanted to check here first what others think and whether this is intentionally left out as a feature.

**Comment by nlohmann** [comment]
We looked into adding a `find(const json_pointer&)` overload for this a while back, and ended up declining it. The blocker is structural: `basic_json::iterator` is bound to its owning container, so an iterator resolved from a nested sub-object throws `invalid_iterator.212` when compared against the root's `end()`. That means such an overload could only ever safely resolve single-token pointers -- for any nested pointer (e.g. `/foo/bar`) it would have to return `end()` unconditionally, regardless of whether the path actually resolves, which is indistinguishable from "not found" and doesn't match what `contains()`/`at()` report for the same pointer. So it wouldn't actually deliver the `contains()`/`at()` symmetry this issue is asking for, and risks being a silent footgun for the common nested-pointer case. For now, `contains(json_pointer)` + `at(json_pointer)` remains the recommended pattern, even though it does mean walking the pointer path twice.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
